tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php

```diff
@@ -14,49 +16,58 @@
 }
 
 class Test_Editor_Loader_Factory extends Elementor_Test_Base {
-	private $original_experiment_default_state;
+	private $original_experiments_states;
 
 	public function setUp(): void {
 		parent::setUp();
 
-		$this->original_experiment_default_state = Plugin::$instance->experiments
-			->get_features( Editor::EDITOR_V2_EXPERIMENT_NAME )['default'];
+		$experiments = Plugin::$instance->experiments;
+
+		$this->original_experiments_states = Collection::make( Editor_V2_Experiments::all() )
+			->map_with_keys( fn ( $name ) => [ $name => $experiments->get_features( $name )['default'] ] )
+			->each( fn ( $state, $name ) => $experiments->set_feature_default_state( $name, Experiments_Manager::STATE_INACTIVE ) );
 	}
 
 	public function tearDown(): void {
 		parent::tearDown();
 
-		Plugin::$instance->experiments
-			->set_feature_default_state( Editor::EDITOR_V2_EXPERIMENT_NAME, $this->original_experiment_default_state );
+		$experiments = Plugin::$instance->experiments;
+
+		$this->original_experiments_states
+			->each( fn ( $state, $name ) => $experiments->set_feature_default_state( $name, $state ) );
 	}
 
 	/**
 	 * @dataProvider create_data_provider
 	 */
-	public function test_create( $state, $query_params, $expected_class ) {
+	public function test_create( $experiment, $state, $expected_class ) {
 		// Arrange
-		Plugin::$instance->experiments->set_feature_default_state(
-			Editor::EDITOR_V2_EXPERIMENT_NAME,
-			$state
-		);
-
-		foreach ( $query_params as $key => $value ) {
-			$_GET[ $key ] = $value;
-		}
+		Plugin::$instance->experiments->set_feature_default_state( $experiment, $state );
 
 		// Act
-		$config_provider = Editor_Loader_Factory::create();
+		$loader = Editor_Loader_Factory::create();
 
 		// Assert
-		$this->assertInstanceOf( $expected_class, $config_provider );
+		$this->assertInstanceOf( $expected_class, $loader );
 	}
 
 	public function create_data_provider() {
```

Comment:
It feels weird to me, maybe we could split the `test_create` into two tests one for when the experiments are active and one for when they are inactive.

```php
/**
	 * @experiments Editor_V2_Experiments::all()
	 */	
	public function test_create_experiment_active( $experimemt ) {		
		Plugin::$instance->experiments->set_feature_default_state( $experiment, 				Experiments_Manager::STATE_ACTIVE);
		$loader = Editor_Loader_Factory::create();

		$this->assertInstanceOf( Editor_V2_Loader::class, $loader );
	}
```

Reply:
but I have multiple experiments

Reply:
IDK, that's the whole point of data providers... 🤷

Reply:
So I don't like them...